src/app/models/function.rb

```diff
@@ -1,5 +1,6 @@
 class Function < ApplicationRecord
   validates :name, presence: true, uniqueness: { scope: :user_id }, format: { with: /\A[0-9a-z_]+[\?!]?\z/i }
+  validates :tags, length: { minimum: 0, maximum: 3, message: :maximum_tags_exceeded }
```

Comment:
I'm puzzled abit, how does it work without `has_and_belongs_to_many :tags` ?

Reply:
It was defined later. I can't see what's wrong with that?

https://github.com/emad-elsaid/rubyfunctions/blob/afd4526ab0463d37fd422ac249fdda3bf365b4c7/src/app/models/function.rb#L16